docs/dev/guide/contract-verification.md

```diff
@@ -0,0 +1,86 @@
+# Contract Verification
+
+## What is contract verification?
+
+The bytecode generated by the Solidity compiler is essentially deployed when a Smart Contract is deployed on a network. The original Solidity code is not retained in any way.
+
+Verifying a contract entails determining whether the on-chain bytecode generated from the Smart Contract source code complies. If it does, we can state that we have confirmed the bytecode's source code's integrity. Hence the name – verification.
+<br>
+
+<hr>
+
+<br>
+
+## How are Smart Contracts verified in zkSync??
+
+The deployed bytecode and the Solidity source code of a Smart Contract are compared during the verification process using zkSync. The algorithm compiles the source code to compare the generated bytecode to the deployed one.
+<br>
+If the two sides match in every way, the contract is verified.
+
+Code verification requires six arguments: 
+
+- The compiler address
+- The contract name
+- The source code, including the source of all imports
+- The version of the compiler used to produce the deployed bytecode
+- Information about the number of compiler optimizations (if any)
+- The constructor arguments
+ 
+ If any of these pieces of information is wrong, the process of verification fails.
+<br>
+
+<hr>
+
+<br>
+
+## Modes of verification
+
+Depending on whether your contract is deployed on a network or with zkSync, there are two modes of contract verification:
+
+- Public: Every zkSync user on the block explorer will be able to check the verified source code and interact with it. 
+- Private: The contract is verified and thus valid within a single project in zkSync, available to you and anybody with access to the project.
```

Comment:
I don't understand this part

Reply:
This section just states that they are generally only two modes of smart contract verification, depending on what network it is verified, however, if the user verifies a smart contract using zkSync block explorer, it's made public to everyone to see and interact with.

Reply:
what is "deployed on a network or with zkSync"?